src/NUnitFramework/framework/Compatibility/System.Collections.Concurrent/ConcurrentQueue.cs

```diff
@@ -0,0 +1,413 @@
+// ConcurrentQueue.cs
+//
+// Copyright (c) 2008 Jérémie "Garuma" Laval
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+//
+
+using System;
+using System.Threading;
+using System.Collections;
+using System.Collections.Generic;
+using System.Runtime.Serialization;
+
+namespace System.Collections.Concurrent
```

Comment:
@CharliePoole I intended to ask the same question. I thought I had typed it, but I guess I forgot :smile: 

I agree with @oznetmaster that it would complicate our interfaces to return two different types, but, I believe that our code only ever adds to the Children and then enumerates over them and adding is only ever done through `AddResult`. Could we drop these two compatibility classes and have our `Children` just be `IEnumerable<ITestResult>` with a backing field of `ConcurrentQueue` on builds that support PARALLEL and the original `List<>` on PCL/SL?

This would allow us to drop these two compatibility classes that aren't really needed anyway since conflicts are not needed.